src/test/java/uk/gov/hmcts/reform/managecase/service/noc/VerifyNoCAnswersServiceTest.java

```diff
@@ -0,0 +1,160 @@
+import uk.gov.hmcts.reform.managecase.service.NoticeOfChangeService;
+
+import javax.validation.ValidationException;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.Mockito.when;
+
+@SuppressWarnings({"PMD.JUnitAssertionsShouldIncludeMessage", "PMD.DataflowAnomalyAnalysis",
+    "PMD.UseConcurrentHashMap"})
+class VerifyNoCAnswersServiceTest {
+
+    @InjectMocks
+    private VerifyNoCAnswersService verifyNoCAnswersService;
+
+    @Mock
+    private NoticeOfChangeService noticeOfChangeService;
+
+    @Mock
+    private ChallengeAnswerValidator challengeAnswerValidator;
+
+    @Mock
+    private PrdRepository prdRepository;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private SearchResultViewItem searchResultViewItem;
+    private ChallengeQuestionsResult challengeQuestionsResult;
+    private CaseViewResource caseViewResource;
+
+    @BeforeEach
+    void setUp() throws JsonProcessingException {
+        MockitoAnnotations.initMocks(this);
+
+        searchResultViewItem = createCase();
+        challengeQuestionsResult = new ChallengeQuestionsResult();
+        caseViewResource = new CaseViewResource();
+        NoCRequestDetails details = NoCRequestDetails.builder()
+            .searchResultViewItem(searchResultViewItem)
+            .challengeQuestionsResult(challengeQuestionsResult)
+            .caseViewResource(caseViewResource)
+            .build();
+
+        when(noticeOfChangeService.challengeQuestions("1")).thenReturn(details);
+    }
+
+    @Test
+    void shouldVerifyNoCAnswersSuccessfully() {
+        mockPrdResponse("ORGID1");
+        when(challengeAnswerValidator.getMatchingCaseRole(Mockito.any(), Mockito.any(), Mockito.any()))
+            .thenReturn("[Defendant]");
+
+        VerifyNoCAnswersRequest request = new VerifyNoCAnswersRequest("1", emptyList());
+
+        NoCRequestDetails result = verifyNoCAnswersService.verifyNoCAnswers(request);
+
+        assertAll(
+            () -> assertThat(result.getOrganisationPolicy().getOrgPolicyCaseAssignedRole(), is("[Defendant]")),
+            () -> assertThat(result.getOrganisationPolicy().getOrgPolicyReference(), is("DefendantPolicy")),
+            () -> assertThat(result.getOrganisationPolicy().getOrganisation().getOrganisationID(), is("QUK822NA")),
+            () -> assertThat(result.getOrganisationPolicy().getOrganisation().getOrganisationName(), is("SomeOrg")),
+            () -> assertThat(result.getSearchResultViewItem(), is(searchResultViewItem)),
+            () -> assertThat(result.getChallengeQuestionsResult(), is(challengeQuestionsResult)),
+            () -> assertThat(result.getCaseViewResource(), is(caseViewResource))
+        );
```

Comment:
Does this need a `verify` to check the interaction with the challengeAnswerValidator ?

Reply:
Since there is no real logic for those values, for the purpose of the test I don't think it adds anything.

Reply:
👍